### PR TITLE
[Slumber] What's New

### PR DESCRIPTION
**Pull request by leandroalonso**
Adds the What's New sheet for the Slumber Studios partnership:

This doesn't add animations and the purchase flow is not complete — I'd like the modal to update if the user completes the purchase.

## To test

1. Go to `WhatsNew.swift` and change the `init` to:

```
self.announcements = announcements
        self.previousOpenedVersion = "7.56"
        self.currentVersion = "7.57"
        self.lastWhatsNewShown = ""
```

2. Go to Profile > Settings > Developer > "Set to No Plus" and Profile > Settings > Beta Features > `tracksLogging`
3. Re-run the app
4. ✅ You should see the What's New modal prompting to subscribe to Plus
5. tap "Subscribe to Plus"
6. ✅ The purchase flow should open, notice that `source: flow` is tracked
7. Dismiss it
8. Tap "Learn more"
9. ✅ A Safari View Controller should open without leaving the app
10. Dismiss everything and go to Profile > Settings > Developer > "Set to Paid Plus"
11. Re-run the app
12. ✅ You should see the modal with the content for a paid user
13. Tap the code (or `****` depending on the value on the server)
14. ✅ The code should be copied to clipboard and a Toast message is shown
15. Tap "Redeem your code"
16. ✅ A page to redeem the code should appear
17. Go to Profile > Settings > Beta Features > `slumber` and disabled it
18. Re-run the app
19. ✅ No What's New is shown

### Purchase flow

1. Re-enable the `slumber` flag
2. Go to Profile > Settings > Developer > "Set to No Plus"
3. Hit <kbd>Command</kbd>+<kbd>Shift</kbd>+<kbd>,</kbd> to open the Scheme editor
4. Click the options tab
5. Select the `Pocket Casts Configuration.storekit` file in the StoreKit configuration dropdown
6. Re-run the app
7. When the What's New appear, tap "Subscribe to plus"
8. Purchase any plan
9. Dismiss the screen
10. ✅ The What's New should update showing the message to a Plus user

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

File: podcasts/Whats New/WhatsNewFullView.swift [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import SwiftUI
+import PocketCastsServer
+
+struct WhatsNewFullView: View {
```
Comment (leandroalonso) [inline podcasts/Whats New/WhatsNewFullView.swift:4]: The goal is for this view to be more generic, but it's a bit specialized on the Slumber announcement due to time constraints. Shall I change the name or keep it?

File: podcasts/Whats New/SlumberWhatsNewHeader.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import PocketCastsServer
+
+struct SlumberWhatsNewHeader: View {
+    var body: some View {
+        HStack {
+            Group {
+                PodcastCover(podcastUuid: "82e37e80-755d-0138-eddc-0acc26574db2")
+                PodcastCover(podcastUuid: "9478cc80-7c42-0138-edfe-0acc26574db2")
+                PodcastCover(podcastUuid: "37082d70-e945-0137-b6eb-0acc26574db2")
+                PodcastCover(podcastUuid: "62200ab0-b7ec-0139-f606-0acc26574db2")
```
Comment (leandroalonso) [inline podcasts/Whats New/SlumberWhatsNewHeader.swift]: I decided to load the covers instead of adding the images to the bundle.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/40">7.57 ❄️</a>. The due date for this milestone has already passed.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/SwiftUI/UnderlineLinkTextView.swift [hunk 3]
```diff
@@ -10,9 +10,10 @@ struct UnderlineLinkTextView: View {
 
     var body: some View {
         // Extract links and add underline to all of them
-        let links = text.matches(for: "(?:__|[*#])|\\[(.*?)\\]\\(.*?\\)")
+        //
+        let links = text.matches(for: "\\[([^\\[\\]]*)\\]\\((.*?)\\)")
```
Comment (leandroalonso) [inline podcasts/SwiftUI/UnderlineLinkTextView.swift]: Regex was wrong and it was matching `**` and `##` while it shouldn't.

**Review by bjtitus — APPROVED** [review]
✅ Everything worked!

I wonder if we could send the code as a query parameter so it doesn't need to be manually copied?

**Comment by leandroalonso** [comment]
> I wonder if we could send the code as a query parameter so it doesn't need to be manually copied?

That's a good question, I'll ask on the project thread.

**Pull request merged**